src/error-handling/error-contexts.md

```diff
@@ -5,25 +5,18 @@ contextual information to your errors and allows you to have fewer
 custom error types:
 
 ```rust,editable,compile_fail
-use std::{fs, io};
+use std::{fs::{self, File}, io};
 use std::io::Read;
-use thiserror::Error;
-use anyhow::{Context, Result};
-
-#[derive(Error, Debug)]
-enum ReadUsernameError {
-    #[error("Found no username in {0}")]
-    EmptyUsername(String),
-}
+use anyhow::{Context, Result, bail};
 
 fn read_username(path: &str) -> Result<String> {
     let mut username = String::with_capacity(100);
-    fs::File::open(path)
+    File::open(path)
         .context(format!("Failed to open {path}"))?
         .read_to_string(&mut username)
         .context("Failed to read")?;
     if username.is_empty() {
-        return Err(ReadUsernameError::EmptyUsername(String::from(path)))?;
+        bail!("Found no username in {}", path);
```

Comment:
I'm assuming `bail!` supports this the same way as `format!` and friends:

```suggestion
        bail!("Found no username in {path}");
```

Reply:
It does, I just don't like it. Done anyway.

Reply:
Yeah, I've been going back and forth on this myself... I was super used to the `"... {} ... {} ...", x, y` style at first: I like how it keeps the format string short and collects everything at the end in a consistent way. Now I'm trying to inline the variables when I can — I find that this typically makes things more readable.

However, it all breaks down and becomes very annoying to look at when you don't have suitably-named variables at hand. When you can either create extra temporaries for the variables (a bit weird) or you can put the expressions at the end of your print call. Mixing the two styles looks really odd to me.

For now, the style used (somewhat consistently, I hope) is to inline the variables.